chapter-1/include/parse.h

```diff
@@ -53,7 +53,7 @@ typedef struct StateTransition {
 } StateTransition;
 
 typedef struct TokenFinder {
-    StateTransition *transitions;
+    const StateTransition *transitions;
     unsigned int transitionCount;
```

Comment:
C question: is there a reason to make the transitions const but not the count?

Reply:
First, just to be clear, the pointer `transitions` is not `const`, but the thing(s) pointed to by `transitions` are (`transitions` is a pointer to `const StateTransition`). It's nice to say this structure references some data, but doesn't own write privileges to it. That alone might be something of an answer to your question.

Why not declare members const? It usually becomes a bit of a pain to initialize:

```
struct TokenFinder *tf = allocate(sizeof(*tf));
tf->transitionCount = 5;       // Won't compile because const
```

Reply:
Makes sense, thanks!